server/src/main/java/org/jboss/as/server/deployment/module/descriptor/DeploymentStructureDescriptorParser.java

```diff
@@ -278,6 +303,17 @@ private void handleDeployment(final DeploymentPhaseContext phaseContext, final D
         }
     }
 
+    private boolean moduleIncluded(final Set<AdditionalModuleSpecification> additionalModules, final ModuleIdentifier dependency) {
+        Iterator<AdditionalModuleSpecification> it = additionalModules.iterator();
+        while (it.hasNext()) {
+            ModuleIdentifier identifier = it.next().getModuleIdentifier();
+            if (identifier.equals(dependency)) {
```

Comment:
The fact we're iterating this makes me wonder: why is it a set?

Reply:
same as replaced Set<ModuleIdentifier> which contains no duplicate elements (module identifier).